### DISCUSSION TITLE
Does anyone have issues rendering a quarto document with MikTeX?

### DISCUSSION THREAD

**Discussion by rccline** [quarto-dev/quarto-cli#9320] [Q&A] [answered]
### Description

Xetex is installed with MikTeX but Quarto keeps looking elsewhere for Xelatex.   xelatex --version runs in my console and identifies the version, but quarto cannot find it.  Is is possible to use MikTex with Quarto?  Should I use some other method to render PDF's in Quarto?  I am lost.  

quarto render "Styled Tables with scales package.qmd" --to pdf
processing file: Styled-Tables-with-scales-package.rmarkdown
output file: Styled-Tables-with-scales-package.knit.md
pandoc
  to: latex
  output-file: Styled-Tables-with-scales-package.tex
  standalone: true
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf

metadata
  documentclass: scrartcl
  classoption:
    - DIV=11
    - numbers=noendperiod
  papersize: letter
  header-includes:
    - '\KOMAoption{captions}{tableheading}'
  block-headings: true
  title: Scales Package
  editor: source
  latex-engine: xelatex

Rendering PDF
running xelatex - 1
command xelatex not found, attempting install
finding package for xelatex
> 1 package to install
> installing xelatex (1 of 1)
ERROR: [non-error-thrown] tlmgr returned a non zero status code
The system cannot find the path specified.

Stack trace:
The system cannot find the path specified.

    at Command.handleError (file:///C:/Users/Robert/AppData/Local/Programs/Quarto/bin/quarto.js:8176:126)
    at Command.parseCommand (file:///C:/Users/Robert/AppData/Local/Programs/Quarto/bin/quarto.js:7901:18)
    at eventLoopTick (ext:core/01_core.js:153:7)
    at async quarto (file:///C:/Users/Robert/AppData/Local/Programs/Quarto/bin/quarto.js:117320:5)
    at async file:///C:/Users/Robert/AppData/Local/Programs/Quarto/bin/quarto.js:117332:9
    at async mainRunner (file:///C:/Users/Robert/AppData/Local/Programs/Quarto/bin/quarto.js:117229:9)
    at async file:///C:/Users/Robert/AppData/Local/Programs/Quarto/bin/quarto.js:117323:5

**Comment by cderv** [comment]
Quarto should work with any LaTeX distribution - it seems you have MikTeX but still tlmgr seems to be used somehow .

Can you run `quarto check` in a terminal from your project and share results ? Thanks. 

Also can you point me to your version and what MikTeX you installed so that I can reproduce ?

**Comment by rccline** [comment]
First let me say: THANK YOU!!  I cannot render in Quarto and now, rendering a PDF in Rmarkdown yields a PDF with tables and headings running through the margins and off the page.  

I have set my system Environmental paths to MikTex
 C:\Users\Robert\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;C:\Program Files\MiKTeX\miktex\bin;

$ quarto check
Quarto 1.5.26
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.26
      Path: C:\Users\Robert\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\Robert\AppData\Roaming\TinyTeX\bin\windows\
      Version: undefined

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.11.1
      Path: C:/Users/Robert/AppData/Local/Programs/Python/Python311/python.exe
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with py -m pip install jupyter

[>] Checking R installation...........OK
      Version: 4.3.3
      Path: C:/PROGRA~1/R/R-43~1.3
      LibPaths:
        - C:/Users/Robert/AppData/Local/R/win-library/4.3
        - C:/Program Files/R/R-4.3.3/library
      knitr: 1.45
      rmarkdown: 2.26

[>] Checking Knitr engine render......OK

I am running MikTeX Console 4.12 
Rstudio 2023.12.1 and R 4.3.3  

 I tried to install MikTex when I created a project to follow a YouTube tutorial.  When I tried to render in PDF, all I found were problems.  I have pushed the Quarto project to:  https://github.com/rccline/stocks

**Reply by cderv** [reply]
````
[>] Checking LaTeX....................OK
Using: TinyTex
Path: C:\Users\Robert\AppData\Roaming\TinyTeX\bin\windows
Version: undefined
````

It seems you have TinyTeX install here (possibly using R when using Rmd ?). So it won't use MikTeX. 
But the version is undefined, so I think this could be leftover from old install. 

You could remove this TinyTeX folder if you don't plan to use it. 

You could also add TinyTeX to your system. It should not conflict with global MikTeX. 

But I understand you want only one environment, should if this is MikTeX you should remove the TinyTeX one.

**Reply by rccline** [reply]
tinytex::uninstall_tinytex()
Error in tinytex::uninstall_tinytex() : 
  TinyTeX does not seem to be installed.  

I have given up on MikTex.  I tried to install TinyTex from https://cran.rstudio.com.  The code runs, but TinyTex is not installed.  The tinytex folder does not exist after running the install:  

install.packages("tinytex", repos = "https://cran.rstudio.com/")
Installing package into ‘C:/Users/Robert/AppData/Local/R/win-library/4.3’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.3/tinytex_0.50.zip'
Content type 'application/zip' length 140750 bytes (137 KB)
downloaded 137 KB

package ‘tinytex’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\Robert\AppData\Local\Temp\Rtmpimeb4q\downloaded_packages
> library(tinytex)
> tinytex::tlmgr_update()
tlmgr update --all --self
Error in system2("tlmgr", args, ...) : '"tlmgr"' not found

**Reply by rccline** [reply]
I was unable to install TinyTex from the Quarto project. I created a project and installed from an .Rproj.  However, 

output file: install_tinytex.knit.md

! Undefined control sequence.
l.134 *C:\Users
               \Robert\AppData\Local\R\win-library\textbackslash4.3\tinytex\R* 

Error: LaTeX failed to compile install_tinytex.tex. See https://yihui.org/tinytex/r/#debugging for debugging tips. See install_tinytex.log for more info.
Execution halted

!  ==> Fatal error occurred, no output PDF file produced!  

**log files from Rproj**  
https://github.com/rccline/stocks/tree/master/Rproj_TinyTex

**Reply by mcanouil** [reply, marked as answer]
See <https://quarto.org/docs/output-formats/pdf-engine.html#installing-tex>.

**Reply by rccline** [reply]
I was not able to render a PDF  with Quarto and MikTex.  **I had to remove the MikTex console before installing TinyTex.**  Thank you for the feedback.

**Reply by cderv** [reply]
@rccline just to explain: WHen you tried to install from R you just installed the R package using `install.packages()`

You would have need to run `tinytex::install_tinytex()` as documented https://yihui.org/tinytex/#for-r-users

Installing with Quarto is another valid option - Both install TinyTeX distribution of TeX Live.
